azerothcore/azerothcore-wotlk
# (db/fix): Shadowglen, Teldrassil Gameobject Improvements

From pkt, many mushrooms, 2 lillies, and 6 missing battered chests added
Based on gathered data, 40 mushrooms and 15 lillies spawn at one time.
1 of the 8 Battered Chests should spawn at a time in Shadowglen
Battered Chests should respawn at an intervals of 1m-5m

Referenced packets in comment lines, uploaded to AC project.
Additional chest timer videos recorded.



## Changes Proposed:
-  Improved Quest Iverron's Antidote Part 2 by increasing the spawns, and pooling them.  Timers used were timers from other objects of the same type.
-  Inserted 6 Battered Chests that were missing from the database, and pooled them together as they should be.  Timers were researched, a fair approximation of Blizzard timers created in the pool.  The original 2 were left as is, they are already in the upper limit of possible timers.

## Issues Addressed:

## SOURCE:
Blizz servers, large assortment of gathered packet data, submitted packets to AC
The packet data is the best data, but I also went onto the mastery seasons PTR to create visual demonstration of the pool and it's approximate spawn times:
https://www.youtube.com/watch?v=W2trN0U18-c
https://www.youtube.com/watch?v=1YLgbDbFBsE
https://www.youtube.com/watch?v=YEgtl6EwOvk

## Tests Performed:

- Checked spawns before pooling
- Checked pool functionality


## How to Test the Changes:


1. Create a character in the NE start area
2. .quest add 3521 so you can interact with the quest objects
3. interact with the quest objects, the spawns should change as they repop (suggest lowering the spawn timers)
4. Check out the 8 battered chests, only 1 should ever be spawned at a time 
![image](https://user-images.githubusercontent.com/91289495/147194503-025bfaa0-1512-48d1-bae7-1c7358ec614d.png)


## Known Issues and TODO List:


- Actual timers for the quest objects prolly can be mined, but it would not be easy and it seems like there is a minimum that exists for at least the lillies.  When I attempted to grab them all up they started respawning faster than I could grab them.
- Battered Chest loot itself is missing at very least the coin, loot tables were not addressed.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
